provider/datagen/src/transform/cldr/source.rs

```diff
@@ -222,3 +227,55 @@ impl<'a> CldrDirLang<'a> {
         }
     }
 }
+
+#[cfg(feature = "icu_transliteration")]
+pub(crate) struct CldrDirTransform<'a>(&'a CldrCache, String);
+
+#[cfg(feature = "icu_transliteration")]
+impl<'a> CldrDirTransform<'a> {
+    pub fn read_and_parse_metadata(
+        &self,
+        transform: &str,
+    ) -> Result<&'a crate::transform::cldr::cldr_serde::transforms::Resource, DataError> {
+        let dir_suffix = self.0.dir_suffix()?;
+        let path = format!("{}-{dir_suffix}/main/{transform}/metadata.json", self.1);
+        if self.0.serde_cache.file_exists(&path)? {
+            self.0.serde_cache.read_and_parse_json(&path)
+        } else {
+            Err(DataErrorKind::Io(std::io::ErrorKind::NotFound)
+                .into_error()
+                .with_display_context(&path))
+        }
+    }
+
+    pub fn read_source(&self, transform: &str) -> Result<String, DataError> {
+        let dir_suffix = self.0.dir_suffix()?;
+        let path = format!("{}-{dir_suffix}/main/{transform}/source.txt", self.1);
+        if self.0.serde_cache.file_exists(&path)? {
+            let buf = self.0.serde_cache.root.read_to_buf(&path)?;
```

Comment:
Change `read_to_buf` by `read_to_string`, then the `AbstractFs` can handle UTF-8 errors. JSON and TOML deserialization can be done on strings as well.

Reply:
Added `read_to_string` to `AbstractFs`